project/src/com/google/daggerqueryui/tests/binding_graph_test.js

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var assert = chai.assert;
+
+describe('Binding Graph', function() {
+  beforeEach(function () {
+    bindingGraph.clear();
+  });
+
+  it('adds one edge correctly', function() {
+    bindingGraph.addEdge('com.google.A', 'com.google.B');
+
+    const expectedNodes = [
+      [0, 'A', 'com.google.A'],
+      [1, 'B', 'com.google.B']
+    ];
+
+    assert.sameDeepMembers(bindingGraph.getNodes().map(node => [node.id, node.label, node.title]), expectedNodes);
```

Comment:
Would it be possible to assert the raw data structure for clarity?

e.g.
```
assert.sameDeepMembers(bindingGraphs.getNodes(), [{id: 0, label: 'A', title: 'com.google.A'}, {id: 1, label: 'B', title: 'com.google.B'}]
```

Reply:
Yes, good idea, thanks
[Done](https://github.com/googleinterns/dagger-query/pull/41/commits/7cdde8a0ba085d2fb5711a791d36da3d0faaef41) 🌱